### PR TITLE
#5 photos upload

### PR DESCRIPTION
**Pull request by Btwol**
#5 migrations fixed, should close

### CODE REVIEW

File: Core/Services/PhotoService.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            string filePath = Path.Combine(uploadsFolder, uniqueFileName);
+            var newPhoto = await _crud.Create<PhotoDto>(new PhotoUploadRequest { Path = filePath, BookId = bookId });
+            using (var stream = System.IO.File.Create(filePath))
+            {
+                await image.CopyToAsync(stream);
+            }
+
+            return new KeyValuePair<int, string>(StatusCodes.Status201Created, "Image added.");
+        }
+        public async Task<KeyValuePair<int, string>> DeletePhoto(int id)
```
Comment (Justyn-98) [inline Core/Services/PhotoService.cs]: Missing empty line between methods.

File: Core/Services/PhotoService.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            _crud = crud;
+            _bookCrud = bookCrud;
+            _mapper = mapper;
+
+            PhotoSizeLimit = long.Parse(configuration["FileSizeLimit"]);
+            AllowedFileExtensions = configuration["AllowedPhotoExtensions"].Split(", ").ToList();
+            uploadsFolder = configuration["PhotoUploadsFolder"];
+        }
+
+        public async Task<KeyValuePair<int,string>> UploadPhoto(IFormFile image, int bookId)
```
Comment (Justyn-98) [inline Core/Services/PhotoService.cs]: For unification use our class ServiceResponse

File: Core/Services/PhotoService.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            _mapper = mapper;
+
+            PhotoSizeLimit = long.Parse(configuration["FileSizeLimit"]);
+            AllowedFileExtensions = configuration["AllowedPhotoExtensions"].Split(", ").ToList();
+            uploadsFolder = configuration["PhotoUploadsFolder"];
+        }
+
+        public async Task<ServiceResponse> UploadPhoto(IFormFile image, int bookId)
+        {
+            if (image == null) return new ErrorResponse() { Message = "No image.", StatusCode = System.Net.HttpStatusCode.BadRequest };
```
Comment (Justyn-98) [inline Core/Services/PhotoService.cs]: Add global "using" for class System.Net.HttpStatusCode.

File: Core/Services/PhotoService.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            if (AllowedFileExtensions.All(ex => extension != ex)) return new ErrorResponse() { Message = "Invalid file extension", StatusCode = System.Net.HttpStatusCode.BadRequest };
+            string uniqueFileName = Guid.NewGuid().ToString() + ".jpeg";
+            string filePath = Path.Combine(uploadsFolder, uniqueFileName);
+            var newPhoto = await _crud.Create<PhotoDto>(new PhotoUploadRequest { Path = filePath, BookId = bookId });
+            using (var stream = System.IO.File.Create(filePath))
+            {
+                await image.CopyToAsync(stream);
+            }
+
+            return new SuccessResponse<PhotoDto>() { Content = newPhoto };
```
Comment (Justyn-98) [inline Core/Services/PhotoService.cs]: Nice to have message like "Photo uploaded successfully" or something like that.

File: WebAPI/Controllers/PhotoController.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+using Storage.Models;
+using Core.Interfaces;
+using Core.DTOs;
+using Core.Requests;
+
+namespace WebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class PhotoController : ControllerBase
```
Comment (Justyn-98) [inline WebAPI/Controllers/PhotoController.cs:24]: You can return ServiceResponse directly in controller. Middleware will catch response and will set Status code, so IActionResult is not neccessary. We want alwas the same response structure in our api.

**Review by wikhag — APPROVED** [review]

**Review by Justyn-98 — APPROVED** [review]

**Pull request merged**